File: android/app/src/main/java/com/slackoff/app/support/EpisodeStore.kt
package com.slackoff.app.support

import android.content.Context
import android.content.SharedPreferences

object EpisodeStore {
    private const val PREFS_NAME = "com.slackoff.episode-selection"
    private var prefs: SharedPreferences? = null

    fun init(context: Context) {
        prefs = context.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE)
    }

    fun selectedEpisode(slug: String): Int? {
        val p = prefs ?: return null
        val value = p.getInt(slug, -1)
        return if (value >= 0) value else null
    }

    fun remember(episode: Int, slug: String) {
        if (slug.isEmpty()) return
        val p = prefs ?: return
        if (p.getInt(slug, -1) == episode) return
        p.edit().putInt(slug, episode).apply()
    }
}

File: SlackOff/Support/EpisodeStore.swift
import Foundation

/// 记录每个视频（slug）当前/上次选中的集数，退出详情页再进来也能恢复。
enum EpisodeStore {
    private static let storageKey = "com.slackoff.episode-selection"

    private static var map: [String: Int] =
        UserDefaults.standard.dictionary(forKey: storageKey) as? [String: Int] ?? [:]

    static func selectedEpisode(forSlug slug: String) -> Int? {
        map[slug]
    }

    static func remember(episode: Int, forSlug slug: String) {
        guard !slug.isEmpty, map[slug] != episode else { return }
        map[slug] = episode
        UserDefaults.standard.set(map, forKey: storageKey)
    }
}